### PR TITLE
[WinCairo] Add memory usage of LibcMalloc category

### PR DESCRIPTION
**Pull request by basuke**
#### 47bfdc5063970f6314c4fc3e86085ff08509ac2e
<pre>
[WinCairo] Add memory usage of LibcMalloc category
<a href="https://bugs.webkit.org/show_bug.cgi?id=239660">https://bugs.webkit.org/show_bug.cgi?id=239660</a>

Reviewed by Ross Kirsling.

WinCairo&apos;s memory timeline just displays JavaScript memory usage. Adding
this info shows also Page
memory usage.

* page/win/ResourceUsageThreadWin.cpp:
(WebCore::ResourceUsageThread::platformCollectMemoryData):

Canonical link: <a href="https://commits.webkit.org/250672@main">https://commits.webkit.org/250672@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@294377">https://svn.webkit.org/repository/webkit/trunk@294377</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Source/WebCore/page/win/ResourceUsageThreadWin.cpp [hunk 1]
```diff
@@ -129,6 +129,14 @@ void ResourceUsageThread::platformCollectMemoryData(JSC::VM* vm, ResourceUsageDa
     data.categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
     data.categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
 
+    usage -= currentGCHeapCapacity;
+    // Following ResourceUsageThreadCocoa implementation
+    auto currentGCOwnedGenerallyInMalloc = currentGCOwnedExtra - currentGCOwnedExternal;
+    if (currentGCOwnedGenerallyInMalloc < mallocBucket.dirtySize)
```
Comment (rkirsling) [inline Source/WebCore/page/win/ResourceUsageThreadWin.cpp]: I think you forgot to change `mallocBucket.dirtySize` to `usage` on this line.

**Review by rkirsling — CHANGES_REQUESTED** [review]

File: Source/WebCore/page/win/ResourceUsageThreadWin.cpp (hunk 1 above)
Comment (basuke) [reply Source/WebCore/page/win/ResourceUsageThreadWin.cpp]: Sorry, that was missing when creating a new branch.

**Review by rkirsling — APPROVED** [review]
LGTM, assuming that it's okay for Windows to base the calculation on totalDirtySize.

**Comment by Constellation** [comment]
@basuke Probably you need to make commit-message formatted.

**Pull request closed without merging**